startstoptorq.sh

```diff
@@ -0,0 +1,275 @@
+# Load the environment
+. ./setenv.sh
+
+# sets the base port for a default TorQ installation
+export KDBHDB=${PWD}/hdb/database
+export KDBWDB=${PWD}/wdbhdb
+export KDBSTACKID="-stackid ${KDBBASEPORT}"
+export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$KDBLIB/l32
+export DEFAULTCSV=${KDBAPPCONFIG}/process.csv
+
+getfield() {
+  fieldno=`awk -F, '{if(NR==1) for(i=1;i<=NF;i++){if($i=="'$2'") print i}}' $csvpath` 		    # get number for field based on headers
+  fieldval=`awk -F, '{if(NR == '$1') print $'$fieldno'}' $csvpath`                    		    # pull one field from one line of file
+  echo $fieldval | envsubst                                                                         # substitute env vars
+ }
+
+parameter() {
+  fieldval=`getfield $1 $2`
+  if [ "" == "$fieldval" ]; then                                                                    # check for empty string
+    echo ""
+  else
+    echo " -"$2 $fieldval
+  fi
+ }
+
+getqcmd() {
+  fieldval=`getfield $1 $2`
+  if [ "" != "$fieldval" ]; then
+    echo $fieldval
+  fi
+ }
+
+findproc() {
+  pgrep -f "\-procname $1 $KDBSTACKID \-proctype $(getfield $1 proctype)"
+ }
+
+startline() {
+  procno=`awk '/'$1'/{print NR}' $csvpath`							    # get line number for file
+  qcmd=`awk '/'$1'/{print NR}' $csvpath`
+  params="proctype U localtime g T w load"  							    # list of params to read from config
+  sline="${TORQHOME}/torq.q -procname $1 ${KDBSTACKID}"					   	    # base part of startup line
+  for p in $params;   										    # iterate over params
+  do
+    a=`parameter $procno $p`;									    # get param
+    sline="$sline$a";     									    # append to startup line
+  done
+  qcmd=`getqcmd $procno "qcmd"`;
+  sline="$qcmd $sline $(getfield $procno extras) -procfile $csvpath"
+  echo $sline
+ }
+
+start() {
+  if [ -z `findproc $1` ]; then									    # check process not running
+    sline="nohup $(startline $1) $extras </dev/null >${KDBLOG}/torq${1}.txt 2>&1 &"                 # line to run each process
```

Comment:
where is `$extras` set? is this a global variable? I think convention in shell scripts is to make global names all caps